### DISCUSSION TITLE
How to enable and enter value in ion-input when ngx-ui-tour is open ?

### DISCUSSION THREAD

**Discussion by kcss-ashvini** [hakimio/ngx-ui-tour#172] [Q&A] [answered]
I have ion-input element to which I have attached the touranchor . when ion-popover is active I'm not able to type anything in the ion-input field

**Comment by hakimio** [comment]
That's because of how [ion-popover](https://ionicframework.com/docs/api/popover) works which is used to display tour step. When using other tour UIs this shouldn't be an issue.
If you can find a simple way to workaround this issue, feel free to submit a PR.

**Comment by kcss-ashvini** [comment]
Also how can I fire a custom click event when the cancel (x) button is pressed? (I want to notify the backend that the tourguide has been skipped ?
Or there another way by which I can give user option to skip the tour and notify backend ?

**Comment by hakimio** [comment]
You can either listen to [the tour events](https://hakimio.github.io/ngx-ui-tour/md-menu/API#events) or provide [custom step template](https://hakimio.github.io/ngx-ui-tour/md-menu/Misc#custom-template) where you can supply your custom `click` listeners.

**Reply by kcss-ashvini** [reply]
But there is no such tour event which is fired when cancel button is pressed

**Reply by hakimio** [reply, marked as answer]
There is `end$` event which you can combine with `stepShow$` to track which step was shown last.
Otherwise, you'll have to implement your own `step template`.

**Reply by kcss-ashvini** [reply]
And is there another way except custome template by which I can change the button title instead of cancel button I want to show skip tour title there ?

**Reply by hakimio** [reply]
You can customize button text using `prevBtnTitle`, `nextBtnTitle`, `endBtnTitle` configs. You can not change the "close" icon button without supplying custom template.

**Reply by kcss-ashvini** [reply]
@hakimio Thank you !

**Reply by JamesGai** [reply]
Hi, could explain further how to use the combination of `end$` and `stepShow$` to enable modification of an input field while tour guide is being activated?

For example if I assign an anchor ID to an input textfield, where current tour guide step shows users where to type something
